### PR TITLE
Support for jpg file

### PR DESCRIPTION
**Pull request by randaz81**
Added support to save files in jpg format. 
Requires libjpeg.
yarpdatadumper updated (--type image_jpg)

### CODE REVIEW

File: src/libYARP_sig/CMakeLists.txt [hunk 1]
```diff
@@ -51,7 +51,12 @@ source_group(TREE "${CMAKE_CURRENT_SOURCE_DIR}"
 set_property(GLOBAL APPEND PROPERTY YARP_TREE_INCLUDE_DIRS "${PROJECT_SOURCE_DIR}/include")
 get_property(YARP_TREE_INCLUDE_DIRS GLOBAL PROPERTY YARP_TREE_INCLUDE_DIRS)
 
-include_directories(${YARP_TREE_INCLUDE_DIRS})
+if (YARP_HAS_JPEG)
+   include_directories(${YARP_TREE_INCLUDE_DIRS} ${JPEG_INCLUDE_DIR})
+else()
+   include_directories(${YARP_TREE_INCLUDE_DIRS})
+endif()
+
```
Comment (drdanz) [inline src/libYARP_sig/CMakeLists.txt]: Please, don't duplicate the code here:
```cmake
include_directories(${YARP_TREE_INCLUDE_DIRS})
if(YARP_HAS_JPEG)
   include_directories(${JPEG_INCLUDE_DIR})
endif()
```

Also, no space after `if`

File: src/libYARP_sig/CMakeLists.txt [hunk 2]
```diff
@@ -64,9 +69,12 @@ add_library(YARP::YARP_sig ALIAS YARP_sig)
...
 
-target_link_libraries(YARP_sig PRIVATE ${ACE_LIBRARIES}
-                               PUBLIC YARP::YARP_conf
-                                      YARP::YARP_OS)
+if (YARP_HAS_JPEG)
+   target_compile_definitions(YARP_sig PRIVATE YARP_HAS_JPEG_C=1)
+   target_link_libraries(YARP_sig PRIVATE ${ACE_LIBRARIES} PUBLIC YARP::YARP_conf YARP::YARP_OS ${JPEG_LIBRARY})
+else()
+   target_link_libraries(YARP_sig PRIVATE ${ACE_LIBRARIES} PUBLIC YARP::YARP_conf YARP::YARP_OS)
+endif()
```
Comment (drdanz) [inline src/libYARP_sig/CMakeLists.txt]: Same here:
```cmake
target_link_libraries(YARP_sig PRIVATE ${ACE_LIBRARIES}
                               PUBLIC YARP::YARP_conf 
                                      YARP::YARP_OS)
if(YARP_HAS_JPEG)
  target_compile_definitions(YARP_sig PRIVATE YARP_HAS_JPEG_C=1)
  target_link_libraries(${JPEG_LIBRARY})
endif()

**Comment by drdanz** [comment]
I think something is missing in the PR, I don't see the actual implementation... Did you forget to add some file?

**Comment by drdanz** [comment]
Please also squash ec1d5152118c268e4b7b8789185bfc7882c3f4d7 into 495e063ab644ea0ab2cd4f6d74dba5316a5bf26a

**Comment by randaz81** [comment]
@pattacini I think libjpeg is already present in icub-common meta-package. It is the same library used by YARP mjpeg carrier. However I'm not 100% sure of this, because libjpeg is a super-standard library and could be available in our system not because of icub-common but because some other library needs it (e.g. Qt?)

For windows:
http://gnuwin32.sourceforge.net/packages/jpeg.htm

**Comment by mbrunettini** [comment]
The `libjpeg-dev` is installed with icub-common not as direct dependency, but requested by other packages (eg. `libopencv-highgui-dev` )
We can add it as a direct dependency in next release, but it's currently installed when` icub-common` is installed

File: src/libYARP_sig/CMakeLists.txt (hunk 2 above)
Comment (traversaro) [reply src/libYARP_sig/CMakeLists.txt]: Nitpick: according to CMake documentation, `JPEG_LIBRARY` is an internal variable and `JPEG_LIBRARIES` should be used instead https://cmake.org/cmake/help/v3.11/module/FindJPEG.html .

**Comment by traversaro** [comment]
For future reference, the relevant `vcpkg` package is `jpeg-turbo`.

**Comment by pattacini** [comment]
Thanks @traversaro for the hint on the name of `vcpkg` port!

**Comment by drdanz** [comment]
Merged, thanks.

**Pull request merged**